mongodb-plugins/src/test/java/co/cask/plugin/etl/test/MongoDBTest.java

```diff
@@ -0,0 +1,282 @@
+    MongoCollection<Document> documents = mongoDatabase.getCollection(MONGO_SINK_COLLECTIONS);
+    Assert.assertEquals(2, documents.count());
+    Iterable<Document> docs = documents.find(new BasicDBObject("ticker", "AAPL"));
+    Assert.assertTrue(docs.iterator().hasNext());
+    for (Document document : docs) {
+      Assert.assertEquals(10, (int) document.getInteger("num"));
+      Assert.assertEquals(500.32, document.getDouble("price"), 0.0001);
+    }
+
+    docs = documents.find(new BasicDBObject("ticker", "CDAP"));
+    Assert.assertTrue(docs.iterator().hasNext());
+    for (Document document : docs) {
+      Assert.assertEquals(13, (int) document.getInteger("num"));
+      Assert.assertEquals(212.36, document.getDouble("price"), 0.0001);
+    }
+  }
+
+  @SuppressWarnings("ConstantConditions")
+  @Test
+  public void testMongoDBSource() throws Exception {
+    ETLStage source = new ETLStage("MongoDB", new ImmutableMap.Builder<String, String>()
+                                              .put(MongoDBBatchSource.Properties.CONNECTION_STRING,
+                                                   String.format("mongodb://localhost:%d/%s.%s",
+                                                                 mongoPort, MONGO_DB, MONGO_SOURCE_COLLECTIONS))
+                                              .put(MongoDBBatchSource.Properties.SCHEMA, SOURCE_BODY_SCHEMA.toString())
+                                              .put(MongoDBBatchSource.Properties.SPLITTER_CLASS,
+                                                   StandaloneMongoSplitter.class.getSimpleName()).build());
+    ETLStage sink = new ETLStage("Table", ImmutableMap.of(Properties.Table.NAME, TABLE_NAME,
+                                                          Properties.Table.PROPERTY_SCHEMA, SINK_BODY_SCHEMA.toString(),
+                                                          Properties.Table.PROPERTY_SCHEMA_ROW_FIELD, "ticker"));
+    ETLBatchConfig etlConfig = new ETLBatchConfig("* * * * *", source, sink, new ArrayList<ETLStage>());
+
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(ETLBATCH_ARTIFACT, etlConfig);
+    Id.Application appId = Id.Application.from(Id.Namespace.DEFAULT, "MongoSourceTest");
+    ApplicationManager appManager = deployApplication(appId, appRequest);
+
+    MapReduceManager mrManager = appManager.getMapReduceManager(ETLMapReduce.NAME);
+    mrManager.start();
+    mrManager.waitForFinish(5, TimeUnit.MINUTES);
+
+    DataSetManager<Table> outputManager = getDataset(TABLE_NAME);
+    Table outputTable = outputManager.get();
+
+    // Scanner to verify the data in the table
+    Scanner scanner = outputTable.scan(null, null);
+    Row row1 = scanner.next();
+    Assert.assertNotNull(row1);
+    scanner.close();
+    Assert.assertArrayEquals("AAPL".getBytes(), row1.getRow());
+    Assert.assertEquals(10, (int) row1.getInt("num"));
+    Assert.assertEquals(23.23, row1.getDouble("price"), 0.00001);
```

Comment:
Yes it is, otherwise conversion of BSONObject of StructuredRecord would have failed. I am adding another test from Mongo to Mongo copy that will provide more confidence.